vm-device/src/interrupt/mod.rs

```diff
@@ -141,6 +141,27 @@ pub trait InterruptSourceGroup: Send + Sync {
     #[allow(unused_variables)]
     fn notifier(&self, index: InterruptIndex) -> Option<EventFd>;
 
+    /// Enable the resamplefd.
+    /// With the resamplefd, vfio INTx interrupt can be automatically unamsk in EOI
+    #[cfg(target_arch = "x86_64")]
+    fn enable_resamplefd(&self, _index: InterruptIndex) -> Result<()> {
+        // Not all interrupt sources need resamplefd.
+        // If not, we can have a no-op here.
+        Ok(())
+    }
+
```

Comment:
Do we need the counter part for `disable_resamplefd()`? If not, why?

Reply:
[https://github.com/torvalds/linux/blob/master/drivers/vfio/pci/vfio_pci_intrs.c]()
```
static void vfio_intx_disable(struct vfio_pci_core_device *vdev)
{
	vfio_virqfd_disable(&vdev->ctx[0].unmask);
	vfio_virqfd_disable(&vdev->ctx[0].mask);
	vfio_intx_set_signal(vdev, -1);
	vdev->irq_type = VFIO_PCI_NUM_IRQS;
	vdev->num_ctx = 0;
	kfree(vdev->ctx);
}
```
The VFIO kernel will stop polling the resamplefd when we call device.disable_irq() from CH. However, the CH user-space IOAPIC may still trigger the resamplefd, although this is harmless. Therefore, adding a disable_resamplefd() function would be more reasonable. I will add this interface and call it from the device.disable_intx() function.

Reply:
> Do we need the counter part for `disable_resamplefd()`? If not, why?

@rbradford I have added the disable_resamplefd() function, which will be called from the device.disable_intx() function.